### PR TITLE
Schema fix for iam policy scope

### PR DESCRIPTION
**Pull request by Vasundharashukla**
1. **What this PR does / why we need it**:
Schema fix for iam policy scope

3. **Additional information**
Make schema types as optional, but mention required in Docs for clarity.

### CODE REVIEW

**Review by ishangupta-ds — DISMISSED** [review]

**Comment by shreyassreenivas** [comment]
Please undo the changes in done in the schema documentation as well. 
and add a line in the resource description stating at least one scope is mandatory

File: docs/resources/iam_policy.md [hunk 1]
```diff
@@ -22,6 +22,10 @@ In the Tanzu Mission Control resource hierarchy, there are three levels and five
 In addition to the direct policy defined for a given object, each object has inherited policies described in the parent objects. For example, a cluster has a direct policy and inherited policies from the cluster group and organization to which it is attached.
 Similarly, a namespace has a direct policy and inherited policies from the workspace with which it is associated.
 
+**Note:**
+The scope parameter is mandatory in the schema and the user needs to add one of the defined scopes to the script
+for the provider to function. At once, the user can define only one scope in one resource.
```
Comment (ishangupta-ds) [inline docs/resources/iam_policy.md:27]: Why is for the provider to function clause in the next line, PTAL?

Also please rephrase the second sentence to -


Only one scope per resource is allowed.

**Review by shreyassreenivas — APPROVED** [review]
LGTM

File: templates/resources/iam_policy.md.tmpl [hunk 2]
```diff
@@ -22,6 +22,10 @@ In the Tanzu Mission Control resource hierarchy, there are three levels and five
 In addition to the direct policy defined for a given object, each object has inherited policies described in the parent objects. For example, a cluster has a direct policy and inherited policies from the cluster group and organization to which it is attached.
 Similarly, a namespace has a direct policy and inherited policies from the workspace with which it is associated.
 
+**Note:**
+The scope parameter is mandatory in the schema and the user needs to add one of the defined scopes to the script
+for the provider to function. At once, the user can define only one scope in one resource.
```
Comment (ishangupta-ds) [inline templates/resources/iam_policy.md.tmpl:27]: Why is for the provider to function clause in the next line, PTAL?

Also please rephrase the second sentence to -

Only one scope per resource is allowed.

**Comment by shreyassreenivas** [comment]
Please ensure that the branch names are descriptional about the changes

**Review by ishangupta-ds — CHANGES_REQUESTED** [review]

**Review by ishangupta-ds — CHANGES_REQUESTED** [review]
Please add the same for security policy resource.

**Pull request closed without merging**